### ISSUE TITLE
Missing Struct when generating from CRD

### ISSUE THREAD

**Issue by Dav1dde** [kube-rs/kopium#84]
The [following CRD](https://github.com/argoproj/argo-cd/blob/v2.4.0/manifests/crds/appproject-crd.yaml) does not generate the struct `AppProjectStatusJwtTokensByRoleItems`, but references it (correctly):

```yaml
          status:
            description: AppProjectStatus contains status information for AppProject
              CRs
            properties:
              jwtTokensByRole:
                additionalProperties:
                  description: JWTTokens represents a list of JWT tokens
                  properties:
                    items:
                      items:
                        description: JWTToken holds the issuedAt and expiresAt values
                          of a token
                        properties:
                          exp:
                            format: int64
                            type: integer
                          iat:
                            format: int64
                            type: integer
                          id:
                            type: string
                        required:
                        - iat
                        type: object
                      type: array
                  type: object
                description: JWTTokensByRole contains a list of JWT tokens issued
                  for a given role
                type: object
            type: object
```

Generates:

```rust
/// AppProjectStatus contains status information for AppProject CRs
#[derive(Serialize, Deserialize, Clone, Debug, JsonSchema)]
pub struct AppProjectStatus {
    /// JWTTokensByRole contains a list of JWT tokens issued for a given role
    #[serde(default, skip_serializing_if = "Option::is_none", rename = "jwtTokensByRole")]
    pub jwt_tokens_by_role: Option<BTreeMap<String, AppProjectStatusJwtTokensByRole>>,
}

/// JWTTokensByRole contains a list of JWT tokens issued for a given role
#[derive(Serialize, Deserialize, Clone, Debug, JsonSchema)]
pub struct AppProjectStatusJwtTokensByRole {
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub items: Option<Vec<AppProjectStatusJwtTokensByRoleItems>>,
}
```

The struct `AppProjectStatusJwtTokensByRoleItems` does not get generated.

To reproduce: `curl -sSL https://raw.githubusercontent.com/argoproj/argo-cd/v2.4.0/manifests/crds/appproject-crd.yaml | kopium -Af -`

Using a fresh `cargo install kopium`

**Comment by clux** [comment]
hm, that definitely looks like a bug. thank you.

presumably another edge case in the [analyzer's array recursion](https://github.com/kube-rs/kopium/blob/main/src/analyzer.rs#L104-L122) with the double `items`, should be a nice test case for this :-)

**Comment by clux** [comment]
Should be fixed in `0.13.0`. Have added [your yaml as a test](https://github.com/kube-rs/kopium/blob/e8cbc912c6051cbc9d073139fed2fa94626790e1/src/analyzer.rs#L957-L1008).

Turned out to be a big oversight in the recursion logic that was hard to spot because 1. it was obscured by big function bodies, and 2. it was unlikely to be hit on smaller CRDs.
